### ISSUE TITLE
=:= failure when a refined type's higher member type has a bound of a higher kind dependent on a type in a closure

### ISSUE THREAD

**Issue by noresttherein** [scala/bug#12742]
Sorry, I have no idea how to describe it better.
## Reproduction steps

Scala version:  2.13.10

```scala	
	type Bound[A]
	trait T[X] {
		type C[-A <: Bound[X]]
		type Of[Z[-_ <: Bound[X]]] = T[X] { type C[-A <: Bound[X]] = Z[A] }
	}

	def m[X, C[-A <: Bound[X]]](t :T[X]#Of[C]) = ???
```

## Problem
```
kinds of the type arguments (C) do not conform to the expected kinds of the type parameters (type Z).
C's type parameters do not match type Z's expected parameters:
type A's bounds <: Playground.Bound[X] are stricter than type _'s declared bounds <: Playground.Bound[X]
	def m[X, C[-A <: Bound[X]]](t :T[X]#Of[C]) = ???
```

It is so  niche I am prepared to hear it's a feature, but to my Bear's Very Little Brain, there is no reason why it should not work, especially that this does:
```Scala
	type Bound[A]
	trait T[X] {
		type C[-A <: Bound[X]]
	}
	type TRefined[X, CT[-A <: Bound[X]]] = T[X] { type C[-A <: Bound[X]] = CT[A] }

	def m[X, C[-A <: Bound[X]]](t :TRefined[X, C]) = ???
```

**Comment by joroKr21** [comment]
I was just going to ask if it works when you don't use a type projection `T[X]#Of[C]` but pass a value of type `T[X]` instead?

**Comment by noresttherein** [comment]
Yes, but the refinement is needed for classes, because you can't use path dependent types on constructor parameter and the fields corresponding to those parameters aren't of the actual argument singleton types anyway. The easiest way is of course just to use refinement directly, but I avoid it like the plague, because, when renaming a member type, IDEs don't rename refinements at use sites. Hence, the 'pattern' of an additional type which hides the refinement, and I like to have it in the outer type itself, rather than the inner object, because it suggests, at the very least, the purpose of the outer type argument, and, once familiar with the convention, suggests that it is a subtype of the outer (projected) type.
